launchdarkly-android-client-sdk/src/main/java/com/launchdarkly/sdk/android/LDClient.java

```diff
@@ -530,6 +530,13 @@ private EvaluationDetail<LDValue> variationDetailInternal(@NonNull String key, @
                     null, defaultValue, false, null);
             result = EvaluationDetail.fromValue(defaultValue, EvaluationDetail.NO_VARIATION, EvaluationReason.error(EvaluationReason.ErrorKind.FLAG_NOT_FOUND));
         } else {
+            if (flag.getPrerequisites() != null) {
+                // recurse on prerequisites to emulate prereq evaluations occurring with desirable side effects such as events for prereqs
+                for (String prereqKey : flag.getPrerequisites()) {
+                    variationDetailInternal(prereqKey, LDValue.ofNull(), false, false);
```

Comment:
From another discussion, this may need to be updated to not go through the hooks code when recursing.

Reply:
Yeah, executing hooks isn't part of the intention and would be in conflict with the hooks spec and the client-side prerequisite events.

Reply:
Being as we don't have hooks in these clients yet, we may want to add a comment in the code to be careful of that when adding hooks.

Reply:
I forgot Android didn't get hooks yet when we were having that discussion.

Added TODO on variationDetailInternal.